feat: add updatePinnedDependencies option (#11087)
renovatebot/renovate: docs/usage/configuration-options.md

To opt in to letting Renovate update internal package versions normally, set this configuration option to true.

## updatePinnedDependencies

> This defaults to `true`. Set to false to disable pinned dependencies updates.

Reviewer: ```suggestion
This defaults to `true`.
Set to `false` to disable pinned dependencies updates.
```

To opt in to letting Renovate update internal package versions normally, set this configuration option to true.

## updatePinnedDependencies

> This defaults to `true`.

Reviewer 2: ```suggestion
By default, Renovate will attempt to update all detected dependencies, regardless of whether they are defined using pinned single versions (e.g. `1.2.3`) or constraints/ranges (e.g. (`^1.2.3`).
```

> Set to `false` to disable pinned dependencies updates.

Reviewer 2: ```suggestion
You can set this option to `false` if you wish to disable updating for pinned (single version) dependencies specifically.
```

This defaults to `true`, meaning that Renovate will perform certain "desirable" updates to _existing_ PRs even when outside of schedule.
If you wish to disable all updates outside of scheduled hours then configure this field to `false`.

## updatePinnedDependencies

> This defaults to `true`.
> Set to false to disable pinned dependencies updates.

Reviewer: ```suggestion
By default, Renovate will attempt to update all detected dependencies, regardless of whether they are defined using pinned single versions (e.g. `1.2.3`) or constraints/ranges (e.g. (`^1.2.3`).
You can set this option to `false` if you wish to disable updating for pinned (single version) dependencies specifically.
```

This is the text that @rarkins wanted to add, but it got "resolved" due to the changes in the markdown file that were from me... 😄

Author: I think in this particular case it was resolved due to me resolving it before pushing the actual commit(I click through them as I apply fixes to not forget anything.)

Author: (It is rare for me to see such online activity on PRs and instead people usually react half a day later when everything is in order already.)
